tests/testWrapper.py

```diff
@@ -0,0 +1,434 @@
+            'SMB Authorization[entry]:SMB login:|||username',
+            'SMB Authorization[password]:SMB password :|||pass',
+            'SNMP Authorization[password]:SNMP Community:some comunity',
+            'SNMP Authorization[entry]:SNMPv3 Username:username',
+            'SNMP Authorization[password]:SNMPv3 Password:pass',
+            'SNMP Authorization[radio]:SNMPv3 Authentication Algorithm:some auth algo',
+            'SNMP Authorization[password]:SNMPv3 Privacy Password:privacy pass',
+            'SNMP Authorization[radio]:SNMPv3 Privacy Algorithm:privacy algo',
+            'ESXi Authorization[entry]:ESXi login name:|||username',
+            'ESXi Authorization[password]:ESXi login password:|||pass']
+        cred_dict = {
+            'ssh': {'type': 'ssh',
+                    'port': '22',
+                    'username': 'username',
+                    'password': 'pass',
+            },
+            'smb': {'type': 'smb',
+                    'username': 'username',
+                    'password': 'pass',
+            },
+            'esxi': {'type': 'esxi',
+                    'username': 'username',
+                    'password': 'pass',
+            },
+            'snmp': {'type': 'snmp',
+                     'username': 'username',
+                     'password': 'pass',
+                     'community': 'some comunity',
+                     'auth_algorithm': 'some auth algo',
+                     'privacy_password': 'privacy pass',
+                     'privacy_algorithm': 'privacy algo',
+            },
+        }
+        self.maxDiff=None
+        ret = w.build_credentials_as_prefs(cred_dict)
+        self.assertEqual(len(ret), len(cred_out))
+        self.assertIn('auth_port_ssh|||22', cred_out)
+
+    def test_build_credentials_ssh_up(self, mock_nvti, mock_db):
+        w =  DummyWrapper(mock_nvti, mock_db)
+        cred_out = [
+            'auth_port_ssh|||22',
+            'SSH Authorization[entry]:SSH login name:|||username',
+            'SSH Authorization[password]:SSH password (unsafe!):|||pass']
+        cred_dict = {
+            'ssh': {'type': 'up',
+                    'port': '22',
+                    'username': 'username',
+                    'password': 'pass',
+            }
+        }
+        self.maxDiff=None
+        ret = w.build_credentials_as_prefs(cred_dict)
+        self.assertEqual(ret, cred_out)
+
+    def test_process_vts(self, mock_nvti, mock_db):
+        vts = {
+            '1.3.6.1.4.1.25623.1.0.100061': {
+                'Data length : ': 'new value',
+                'Do not randomize the  order  in  which ports are scanned': 'new value'},
+            'vt_groups': ['family=debian', 'family=general']
+        }
+        vt_out = ['1.3.6.1.4.1.25623.1.0.100061'], [['Mantis Detection[checkbox]:Do not randomize the  order  in  which ports are scanned', 'new value'], ['Mantis Detection[entry]:Data length : ', 'new value']]
```

Comment:
```suggestion
        vt_out = (
            ['1.3.6.1.4.1.25623.1.0.100061'],
            [['Mantis Detection[checkbox]:Do not randomize the  order  in  which ports are scanned', 'new value'],
            ['Mantis Detection[entry]:Data length : ', 'new value']]
        )
```

Reply:
I am not sure if i did get the content correct here. vt_out does contain a list of list of lists :confused: